### PR TITLE
feat: allow `FormData` bodies in `request`

### LINKED ISSUE #1330: feature request: allow sending FormData in `undici.request`?
## This would solve...

In a codebase I am working on, it is a requirement to be able to send FormDatas to the server (to upload files, etc.). It is preferable to use `request` because it is currently much faster than `fetch`.

## The implementation should look like...

Undici's fetch currently parses the FormData to a stringified version:
https://github.com/nodejs/undici/blob/c0961f4c7ef5dd6e64dbb7ffbd3f6cb1f78f106a/lib/fetch/body.js#L75-L124

## I have also considered...

Currently I have an implementation identical to this that just removed the async generator and manually sets the body & header properly. This is very tedious, but it does work.

## Additional context

This is basically all that is missing from making a seamless transition from `node-fetch` to `undici.fetch` to `undici.request`.

### PR DESCRIPTION
**Pull request by KhafraDev**
Fixes #1330 

Allows sending a FormData body to `undici.request`.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -1580,10 +1586,26 @@ async function writeIterable ({ body, client, request, socket, contentLength, he
     .on('close', onDrain)
     .on('drain', onDrain)
 
+  let bodyAsyncIterable = body
+
+  if (util.isFormDataLike(body)) {
+    if (nodeMajor < 16 || (nodeMajor > 16 && nodeMinor < 5)) {
+      throw new InvalidArgumentError('Form-Data bodies are only supported in node v16.5 and newer.')
+    }
```
Comment (almeidx) [inline lib/client.js:1594]: This will throw on 17.5, 18.5, 19.5, etc.

Comment (KhafraDev) [reply lib/client.js]: good catch, fixed in 5169b98022b20c8dfc16fafdcaa707636e2df051

**Comment by mcollina** [comment]
go ahead and add busboy!

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by cjh980402** [comment]
Although it has already been merged, it seems a better way to use `process.version.node` instead of `process.version.slice(1)`.

**Pull request merged**